app/gilded-rose.ts

```diff
@@ -1,69 +1,89 @@
 export class Item {
-  name: string;
-  sellIn: number;
-  quality: number;
-
-  constructor(name, sellIn, quality) {
-    this.name = name;
-    this.sellIn = sellIn;
-    this.quality = quality;
-  }
+    name: string;
+    sellIn: number;
+    quality: number;
+
+    constructor(name, sellIn, quality) {
+        this.name = name;
+        this.sellIn = sellIn;
+        this.quality = quality;
+    }
 }
 
 export class GildedRose {
-  items: Array<Item>;
-
-  constructor(items = [] as Array<Item>) {
-    this.items = items;
-  }
-
-  updateQuality() {
-    for (let i = 0; i < this.items.length; i++) {
-      if (this.items[i].name != 'Aged Brie' && this.items[i].name != 'Backstage passes to a TAFKAL80ETC concert') {
-        if (this.items[i].quality > 0) {
-          if (this.items[i].name != 'Sulfuras, Hand of Ragnaros') {
-            this.items[i].quality = this.items[i].quality - 1
-          }
+    items: Array<Item>;
+
+    constructor(items = [] as Array<Item>) {
+        this.items = items;
+    }
+
+    increaseQuality(item, increment = 1) {
+        item.quality = Math.min(item.quality + increment, 50)
+    }
+
+    decreaseQuality(item, decrement = 1) {
+        item.quality = Math.max(0, item.quality - decrement)
+    }
+
+    dealWithAgedBrie(agedBrie) {
```

Comment:
Not a big fan of using `dealWith` to describe what a function does - maybe `increaseQualityOfAgedBrie` is a more descriptive name? Same goes for the other functions